dialogue-core/src/main/java/com/palantir/dialogue/core/DialogueChannel.java

```diff
@@ -207,7 +218,33 @@ public DialogueChannel build() {
                     .build();
             createMeter.mark();
 
-            return new DialogueChannel(cf, channelFactory, stickySessionFactory);
+            ImmutableList<Channel> perHostChannels = IntStream.range(0, channels.size())
+                    .mapToObj(index -> new ChannelAndFactory() {
+
+                        private final LimitedChannel stickyLimitedChannel =
+                                StickyConcurrencyLimitedChannel.createForQueueKey(
+                                        nodeSelectionChannel, cf.channelName());
+                        private final Channel queueOverride =
+                                QueuedChannel.createPerHost(cf, stickyLimitedChannel, index);
+
+                        @Override
+                        public EndpointChannel endpoint(Endpoint endpoint) {
+                            EndpointChannel endpointChannel = channelFactory.endpoint(endpoint);
+                            return request -> {
+                                request.attachments().put(QueueAttachments.QUEUE_OVERRIDE, queueOverride);
+                                nodeSelectionChannel.routeToHost(index, request);
+                                return endpointChannel.execute(request);
+                            };
+                        }
+
+                        @Override
+                        public ListenableFuture<Response> execute(Endpoint endpoint, Request request) {
+                            return endpoint(endpoint).execute(request);
+                        }
+                    })
+                    .collect(ImmutableList.toImmutableList());
```

Comment:
I think this piece is more or less the alternative described [here](https://github.com/palantir/dialogue/pull/1334#discussion_r671334170), except that we don't leverage it from the StickySessionFactory.

Reply:
Yeah: I think it's valuable to provide wholesale fairness implementation. If we're not going to ship the whole thing (i.e. you can support internal usecase for per-transaction fairness, with ability to create a new sticky channel from an existing sticky channel with new queue), I'd be happy to switch to per-host queues even for sticky channels. The reason I added it in https://github.com/palantir/dialogue/pull/1334 is because I wanted to ship the whole thing.